### ISSUE TITLE
conflicting dependencies kubernetes-cni-0.8.7

### ISSUE THREAD

**Issue by IKatsu** [kubernetes/release#2099]
This is on a fresh install (last week from current time) on RHEL8.3
Running 'dnf update'

 Problem: cannot install both kubelet-1.18.4-0.x86_64 and kubelet-1.21.1-0.x86_64
  - cannot install the best update candidate for package kubernetes-cni-0.8.7-0.x86_64
  - cannot install the best update candidate for package kubelet-1.21.1-0.x86_64

# rpm -qa | grep -i kube
kubernetes-cni-0.8.7-0.x86_64
kubelet-1.21.1-0.x86_64
kubectl-1.21.1-0.x86_64
kubeadm-1.21.1-0.x86_64

Somehow there is a dependency in those packages that wants to downgrade back to kubelet 1.18 when doing 'dnf update'

Red Hat Enterprise Linux release 8.3 (Ootpa)
Kernel 4.18.0-240.22.1.el8_3.x86_64 #1 SMP Thu Mar 25 14:36:04 EDT 2021 x86_64 x86_64 x86_64 GNU/Linux

Using CentOS 7 repo for kubernetes.

**Comment by saschagrunert** [comment]
Thank you for the report, @IKatsu. Do you think you can share an encapsulated reproducer?

PTAL @kubernetes/release-engineering

**Comment by IKatsu** [comment]
I followed this guide: https://swapnasagarpradhan.medium.com/install-a-kubernetes-cluster-on-rhel8-with-conatinerd-b48b9257877a
It's a minimal install RHEL8.3 fully updated, then go through the kubernetes install and type 'dnf update' and it will error.

That's all there is really.
I just installed a 3 node test cluster and all nodes show this behaviour.

**Comment by neolit123** [comment]
that is not an officially supported k8s guide.

have you tried following the steps from here?
https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#installing-kubeadm-kubelet-and-kubectl

**Comment by IKatsu** [comment]
I don't think it matters from which guide you copy/paste 'dnf install -y kubelet kubeadm kubectl'

The only difference is the exclude in the repo file but excluding kubernetes packages from updating without disabling excludes just hides the problem till you actually want to update.

**Comment by neolit123** [comment]
> The only difference is the exclude in the repo file but excluding kubernetes packages from updating without disabling excludes just hides the problem till you actually want to update.

the official guide has no `dnf intall`, it has `yum install`.
https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#installing-kubeadm-kubelet-and-kubectl

i do agree that potential conflicts should be resolved, but is the problem reproducible with `yum install`?

**Comment by shawnz** [comment]
@neolit123 yum is simply a simlink to dnf on RHEL8-based systems. I invoke the command with yum and see the same conflict in which it tries to install version 1.18.4 if the desired version is not explicitly specified during updates.
